app/models/name.rb

```diff
@@ -439,7 +439,14 @@ def real_search_name
   def self.display_to_real_text(name)
     name.display_name.gsub(/ ^\*?\*?__ | __\*?\*?[^_\*]*$ /x, "").
       gsub(/__\*?\*? [^_\*]* \s (#{ANY_NAME_ABBR}) \s \*?\*?__/x, ' \1 ').
-      gsub(/__\*?\*? [^_\*]* \*?\*?__/x, " ") # (this part should be unnecessary)
+      gsub(/__\*?\*? [^_\*]* \*?\*?__/x, " "). # (this part should be unnecessary)
+      # Because "group" was removed by the 1st gsub above,
+      # tack it back on (if it was part of display_name)
+      concat(group_suffix(name))
```

Comment:
Don't worry, display_to_real_text is already an ugly kludge to fix something I couldn't figure out how to fix "correctly" years ago.  Some day in the distant future when we have nothing better to do -- ha! -- we can revise the whole process of parsing and multiply-formatting names.